controllers/k8ssandra/datacenters.go

```diff
@@ -63,13 +63,26 @@ func (r *K8ssandraClusterReconciler) reconcileDatacenters(ctx context.Context, k
 		// its fields are pointers, and without the copy we could end of with shared
 		// references that would lead to unexpected and incorrect values.
 		dcConfig := cassandra.Coalesce(cassClusterName, kc.Spec.Cassandra.DeepCopy(), dcTemplate.DeepCopy())
+
+		// Ensure we have a valid PodTemplateSpec before proceeding to modify it.
+		if dcConfig.PodTemplateSpec == nil {
+			dcConfig.PodTemplateSpec = &corev1.PodTemplateSpec{}
+			// we need to declare at least one container, otherwise the PodTemplateSpec struct will be invalid
```

Comment:
Why can't PodTemplateSpec stay as nil? PodTemplateSpec is not required by cass-operator.

Reply:
We always customize things in K8ssandra-operator and always rely on the podTemplateSpec.
There's only one very edgy case where it's not true, which is when auth is disabled, medusa is disabled and metrics filters are explicitly voided with an empty list.
Doing this will simplify a lot future developments on K8ssandra-operator.